addons/web/static/src/views/fields/progress_bar/progress_bar_field.js

```diff
@@ -73,7 +73,11 @@ export class ProgressBarField extends Component {
     onCurrentValueChange(ev) {
         let parsedValue;
         try {
-            parsedValue = parseFloat(ev.target.value);
+            if (this.props.type === 'integer') {
+                parsedValue = parseInteger(ev.target.value);
+            } else {
+                parsedValue = parseFloat(ev.target.value);
+            }
```

Comment:
Has this been discussed with the framework team? It's a slight change in behavior

Reply:
> Has this been discussed with the framework team? It's a slight change in behavior

no discussion with framework team about this. you can do